PhysicsTools/NanoAOD/python/jets_cff.py

```diff
@@ -76,6 +76,8 @@
 run2_jme_2016.toModify( tightJetIdLepVeto.filterParams, version = "WINTER16" )
 run2_jme_2017.toModify( tightJetId.filterParams, version = "WINTER17" )
 run2_jme_2017.toModify( tightJetIdLepVeto.filterParams, version = "WINTER17" )
+run2_nanoAOD_102Xv1.toModify( tightJetId.filterParams, version = "SUMMER18" )
+run2_nanoAOD_102Xv1.toModify( tightJetIdLepVeto.filterParams, version = "SUMMER18" )
```

Comment:
the run2_nanoAOD_102Xv1 is for the EOY 2018. 
you likely want to do this also for the EOY 2017 and EOY 2016
'run2_nanoAOD_94XMiniAODv1'
'run2_nanoAOD_94XMiniAODv2'
'run2_nanoAOD_94X2016'

same below for puppiJetID

Reply:
Is the  EOY16-27 nano taken care already by run2_jme_2016 and run2_jme_2017 ?

Reply:
this is what I assumed from the existing code. better if @peruzzim could confirm?
is "run2_jme_2016" and "run2_jme_2017" always used in combination with
'run2_nanoAOD_94XMiniAODv1' or 'run2_nanoAOD_94XMiniAODv2' or 'run2_nanoAOD_94X2016'
(I assume otherwise the existing setup wouldn't work)
Maybe better rename them to run2_jme_94X2017 and run2_jme_94X2016?
For UL we will not need "run2_jme_2016" and "run2_jme_2017" anymore since all 3 years will use the same jet ID.

Reply:
The run2_jme_2016 and the run2_jme_2017 are loaded in the general Run2_2016 and Run2_2017.
Nothing to do with the nano modifiers.
https://github.com/cms-sw/cmssw/blob/master/Configuration/Eras/python/Era_Run2_2016_cff.py#L16-L17
https://github.com/cms-sw/cmssw/blob/master/Configuration/Eras/python/Era_Run2_2017_cff.py#L26-L28

to my understanding happens the following:

For Run2_2016,run2_miniAOD_80XLegacy --> pick the "WINTER16" 
(old 80x samples those are data (07Aug17) and RunIISummer16MiniAODv2 MC) 

For Run2_2016,run2_nanoAOD_94X2016 -->  pick the "WINTER16" 
( this is EOY -- those are 2016 94X data (17Jul2018) and RunIISummer16MiniAODv3 MC) 

For  Run2_2016,run2_nanoAOD_106Xv1 --> pick the "WINTER16" 
( this is the current legacy , If you want explicitely this as SUMMER18 you need to specify it)

same for the UL17:

For Run2_2017,run2_nanoAOD_94XMiniAODv1 (EOY) -->  pick the "WINTER17" 
For Run2_2017, run2_nanoAOD_94XMiniAODv2 (EOY) -->  pick the "WINTER17" 
For Run2_2017, run2_nanoAOD_106Xv1  (UL) --> pick the "WINTER17" 
( this is the current legacy , If you want explicitely this as SUMMER18 you need to specify it)

Reply:
Thanks, I've adapted the logic, not making use of the year-modifiers anymore, since they don't hold for UL anymore.

Reply:
very clear 👍